pytest_inmanta/plugin.py

```diff
@@ -1130,3 +1142,16 @@ def finalize_handler(self, handler: ResourceHandler) -> None:
     def finalize_all_handlers(self) -> None:
         for handler_instance in self._handlers:
             self.finalize_handler(handler_instance)
+
+
+@pytest.fixture(scope="session")
+def session_temp_dir(tmpdir_factory: "TempdirFactory") -> Generator[str, None, None]:
+    session_temp_dir = tmpdir_factory.mktemp("session")
+    yield str(session_temp_dir)
+    session_temp_dir.remove(ignore_errors=True)
+
+
+@pytest.fixture
+def use_session_temp_dir(session_temp_dir: str) -> Generator[str, None, None]:
```

Comment:
* Why do we keep the same `state_dir` for the entire session. Couldn't that cause issues?
* I think this implementation could cause issues for the `lsm` module. The `lsm` module often uses the `project` fixture of `pytest-inmanta` together with the `server` fixture of `pytest-inmanta-extensions`. Both fixtures set the `state_dir` (See: https://github.com/inmanta/inmanta-core/blob/ed50df1bbd19f28ee2716bec1272d4237b5e0f92/tests/conftest.py#L675). The following scenario might happen for a test case that uses the `server` and the `project` fixture:
    * The server fixture activates, it sets the state_dir and it starts the server
    * The project fixture activates and it sets the state_dir to a new value as well
    * Depending on the implementation of the server part of the state is now written to the state_dir set by the server fixture and some other part to the state_dir set by the project fixture. I don't think that's desired. I am going to add @wouterdb as a reviewer as well to get his opinion on this. One possible solution would be to only update the state_dir when it's current value matches the default value. That would work, but it's still not an elegant solution.